src/kmipapi/kmip14.go

```diff
@@ -275,9 +286,124 @@ func (kmips *kmip14service) RevokeKey(ctx context.Context, settings *Configurati
 	return &RevokeKeyResponse{UniqueIdentifier: uid}, nil
 }
 
-// Register: Not Implemented
-func (kmips *kmip14service) Register(ctx context.Context, settings *ConfigurationSettings, req *RegisterRequest) (*RegisterResponse, error) {
-	return &RegisterResponse{}, fmt.Errorf("command is not implemented")
+// Register: Register a key
+func (kmips *kmip14service) RegisterKey(ctx context.Context, settings *ConfigurationSettings, req *RegisterKeyRequest) (*RegisterKeyResponse, error) {
+
+	logger := klog.FromContext(ctx)
+	logger.V(4).Info("====== register key ======")
+
+	var err error
+	var decoder *ttlv.Decoder
+	var item *kmip.ResponseBatchItem
+
+	//newkey := hex.EncodeToString([]byte(req.KeyMaterial))
+	newkey := []byte(req.KeyMaterial)
+	payload := kmip.RegisterRequestPayload{}
+
+	if req.Type == "SecretData" && req.DataType == "Password" && req.KeyFormat == "Opaque" && req.KeyMaterial != "" {
```

Comment:
```suggestion
```

Reply:
If the caller passes in any value that does not match these tests, then err will be nil and this function will fail. I suggest removing this if statement and sending the command to the KMIP Server and let it pass or fail the command.